From 01c6f07eb729ad9fa6f2e353a18b0f330ffb2982 Mon Sep 17 00:00:00 2001
From: Yuxiao Qu <yuxiao1155@gmail.com>
Date: Mon, 17 Jul 2023 10:25:28 -0500
Subject: [PATCH] [XrdHttp] Add back parsing of Transfer-Encoding header

This commit reintroduces a header parser for Transfer-Encoding: chunked to maintain compatibility with the EGI Nagios probe. This issue was first mentioned in #2058.

Fixes: d96b2b70487e159b47c24925abcfa00f975ec3d6, #2009
---
 src/XrdHttp/XrdHttpReq.cc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/XrdHttp/XrdHttpReq.cc b/src/XrdHttp/XrdHttpReq.cc
index e11918171..c1d134951 100644
--- a/src/XrdHttp/XrdHttpReq.cc
+++ b/src/XrdHttp/XrdHttpReq.cc
@@ -186,6 +186,8 @@ int XrdHttpReq::parseLine(char *line, int len) {
       sendcontinue = true;
     } else if (!strcasecmp(key, "TE") && strstr(val, "trailers")) {
       m_trailer_headers = true;
+    } else if (!strcasecmp(key, "Transfer-Encoding") && strstr(val, "chunked")) {
+      m_transfer_encoding_chunked = true; 
     } else if (!strcasecmp(key, "X-Transfer-Status") && strstr(val, "true")) {
       m_transfer_encoding_chunked = true;
       m_status_trailer = true;
-- 
2.34.1

